governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java

```diff
@@ -97,9 +112,10 @@ public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
                     try {
                         annotatedFields.add(selfClass().getDeclaredField(name));
                         break;
-                    } 
-                    catch (NoSuchFieldException e) {
-                        throw new IllegalStateException(e);
+                    } catch (NoSuchFieldException e) {
+                        throw new IllegalStateException("Error visiting field " + name, e);
+                    } catch (NoClassDefFoundError e) {
+                        log.info("Unable to scan field '{}' of class '{}'", name, selfClass().getName(), e.getMessage());
```

Comment:
Thanks, this will allow us to create lazy singleton providers with optional dependencies to reduce transitive dependency footprint while still fully participating in DI framework object management.

The addition of class name to the info message should be really helpful for people investigating unexpected behaviors. Would class name also be useful in the NoSuchFieldException message?